omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptReferenceTermSearchHandler1_9.java

```diff
@@ -75,84 +77,116 @@ public PageableResult search(RequestContext context) throws ResponseException {
 			throw new InvalidSearchException("Invalid searchType parameter");
 		}
 
-		ConceptSource conceptSource = conceptService.getConceptSourceByUuid(source);
-
-		if (conceptSource == null) {
-			conceptSource = conceptService.getConceptSourceByName(source);
-		}
-
-		if (conceptSource == null) {
-			return new EmptySearchResult();
+		if (searchType == null) {
+			searchType = "equals";
 		}
 
-		if (code != null || name != null) {
-			if(SEARCH_TYPE_STARTS_WITH.equals(searchType)){
-				return searchStartsWith(context, code, name, conceptSource);
-			} else {
+		if (searchType.equals(SEARCH_TYPE_EQUALS)) {
+			ConceptSource conceptSource = conceptService.getConceptSourceByUuid(source);
+			if (conceptSource == null) {
+				conceptSource = conceptService.getConceptSourceByName(source);
+                if (conceptSource == null) {
+                    return new EmptySearchResult();
+                }
+			}
+			if (code != null || name != null) {
 				return searchEquals(context, code, name, conceptSource);
 			}
+			else {
+				List<ConceptReferenceTerm> terms = conceptService.getConceptReferenceTerms(null, conceptSource,
+						context.getStartIndex(), context.getLimit(), context.getIncludeAll());
+				int count = conceptService.getCountOfConceptReferenceTerms(null, conceptSource, context.getIncludeAll());
+				boolean hasMore = count > (context.getStartIndex() + context.getLimit());
+				return new AlreadyPaged<ConceptReferenceTerm>(context, terms, hasMore);
+			}
 		} else {
-			List<ConceptReferenceTerm> terms = conceptService.getConceptReferenceTerms(null, conceptSource,
-					context.getStartIndex(), context.getLimit(), context.getIncludeAll());
-			int count = conceptService.getCountOfConceptReferenceTerms(null, conceptSource, context.getIncludeAll());
-			boolean hasMore = count > (context.getStartIndex() + context.getLimit());
-
-			return new AlreadyPaged<ConceptReferenceTerm>(context, terms, hasMore);
+			List<ConceptSource> conceptSources = new ArrayList<ConceptSource>();
+			ConceptSource singleConceptSource = conceptService.getConceptSourceByUuid(source);
+			if (singleConceptSource != null) {
+				conceptSources.add(singleConceptSource);
+			}
+			else {
+				List<ConceptSource> allConceptSources = conceptService.getAllConceptSources();
+				if (allConceptSources.size() == 0) {
+					return new EmptySearchResult();
+				}
+				else {
+					for (ConceptSource conceptSource : allConceptSources) {
+						if (conceptSource.getName().contains(source)) {
+							conceptSources.add(conceptSource);
+						}
+					}
+				}
+			}
+			return searchStartsWith(context, code, name, conceptSources);
 		}

```

Comment:
Please throw IllegalStateExcepiton if searchType is neither equals nor startsWith.